File: leetcode/1/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = dict()        
        for i in range(0, len(nums)):
            a = nums[i]
            b = target - a
            if a in d:
                return [d[a] ,i]
            else:
                d[b] = i
            
        

File: leetcode/1/Solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++)  {
            Integer pos = map.get(target - nums[i]);
            if (pos == null) {
                map.put(nums[i], i);    
            } else {
                return new int[]{pos, i};
            }
        }
        return null;
    }
}
